ui.tests/test-module/specs/tabsontop.spec.js

```diff
@@ -0,0 +1,133 @@
+    dropComponent(responsiveGridDropZoneSelector, "Adaptive Form Tabs On Top component", afConstants.components.forms.resourceType.tabsontop);
+  }
+
+  const dropTextInputInTabComponent = function() {
+    const responsiveGridDropZoneSelector = sitesSelectors.overlays.overlay.component + "[data-text='Please drag Tab components here']:last";
+    dropComponent(responsiveGridDropZoneSelector, "Adaptive Form Text Box", afConstants.components.forms.resourceType.formtextinput);
+  }
+
+  const dropTabsInSites = function() {
+    const dataPath = "/content/core-components-examples/library/adaptive-form/panelcontainer/jcr:content/root/responsivegrid/demo/component/container/*",
+        responsiveGridDropZoneSelector = sitesSelectors.overlays.overlay.component + "[data-path='" + dataPath + "']";
+    dropComponent(responsiveGridDropZoneSelector, "Adaptive Form Tabs On Top component", afConstants.components.forms.resourceType.tabsontop);
+  }
+
+  const testPanelBehaviour = function(tabsEditPathSelector, tabsContainerDrop, isSites) {
+    if (isSites) {
+      dropTabsInSites();
+    } else {
+      dropTabsInContainer();
+    }
+    cy.openEditableToolbar(sitesSelectors.overlays.overlay.component + tabsEditPathSelector);
+    cy.invokeEditableAction("[data-action='CONFIGURE']"); // this line is causing frame busting which is causing cypress to fail
+    // Check If Dialog Options Are Visible
+    cy.get("[name='./name']")
+    .should("exist");
+    cy.get("[name='./jcr:title']")
+    .should("exist");
+    cy.get("[name='./dataRef']")
+        .should("exist");
+    cy.get("[name='./visible']")
+        .should("exist");
+    cy.get("[name='./enabled']")
+        .should("exist");
+    cy.get("[name='./assistPriority']")
+        .should("exist");
+    cy.get("[name='./custom']")
+        .should("exist");
+    cy.get("[name='./minItems']")
+        .should("exist");
+    cy.get("[name='./maxItems']")
+        .should("exist");
+
+    cy.get('.cq-dialog-cancel').click();
+    cy.deleteComponentByPath(tabsContainerDrop);
+  }
+
+  context('Open Forms Editor', function() {
+    const pagePath = "/content/forms/af/core-components-it/blank",
+        tabsPath = pagePath + afConstants.FORM_EDITOR_FORM_CONTAINER_SUFFIX + "/tabsontop",
+        panelContainerPathSelector = "[data-path='" + tabsPath + "']";
+    beforeEach(function () {
+      // this is done since cypress session results in 403 sometimes
+      cy.openAuthoring(pagePath);
+    });
+
+    it('insert Tabs on top in form container', function () {
+      dropTabsInContainer();
+      cy.deleteComponentByPath(tabsPath);
+    });
+
+    it('drop element in tabs on top', function () {
+        dropTabsInContainer();
+        dropTextInputInTabComponent();
+        cy.get(`[data-path="${tabsPath}"] [data-path="${tabsPath}/textinput"]`).should('be.visible');
+        cy.deleteComponentByPath(tabsPath);
+    });
+
+    it ('open edit dialog of Tab on top', function(){
```

Comment:
test case for switching tabs in editor (thats dependent on editor message channel) missing.

Reply:
added